### PR TITLE
SendGrid WebApi implementation for sending mail in production environment

### PR DESCRIPTION
**Pull request by juliepromact**
Implemented Sendgrid WebAPI as per issue #96
 
**Files Changed:**

**StringConstant.cs** – Added string constants.
**AppSettingUtil.cs**  - Removed unnecessary usings.
**AppSettings.cs**  - Added SendGridApi property
**UserRepository.cs**  - Uncommented SendMail method and removed logging.
**appsettings.development.example.json**  - Added SendGridApi example.
**project.json**  - Added package SendGrid.NetCore
**SendGridEmailSender.cs** - Implementation added for sending mail in production.
**StartUP.cs** - Added resolving implementation of IEmailSender based on environment(development or production)

### CODE REVIEW

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Startup.cs [hunk 1]
```diff
@@ -27,9 +27,10 @@ namespace Promact.Oauth.Server
     public class Startup
     {
         private ILoggerFactory _loggerFactory { get; }
-
+        private IHostingEnvironment CurrentEnvironment { get; set; }
```
Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Startup.cs]: Why do you need this?  IHostingEnvironment is already injected in the required method

Comment (juliepromact) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Startup.cs]: IHostingEnvironment is not injected in ConfigureServices method

Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Startup.cs]: public Startup(IHostingEnvironment env, ILoggerFactory loggerFactory)

I think It is

Comment (juliepromact) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Startup.cs]: Yes. But ConfigureServices is where we have done the resolving of implementations of IEmailSender.cs
  if (CurrentEnvironment.IsDevelopment())
                services.AddTransient<IEmailSender, AuthMessageSender>();
            else if (CurrentEnvironment.IsProduction())
                services.AddTransient<IEmailSender, SendGridEmailSender>();

Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Startup.cs]: Ok, in that case, naming convention is wrong. private variables are always camel cased with underscore.

Comment (juliepromact) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Startup.cs]: Ok. Will change it.

**Pull request merged**